pkgs/development/tools/just/default.nix

```
@@ -20,22 +20,50 @@ rustPlatform.buildRustPackage rec {
     export USER=just-user
     export USERNAME=just-user
 
-    sed -i tests/integration.rs \
+    sed -i tests/*.rs \
         -e "s@/bin/echo@${coreutils}/bin/echo@g" \
         -e "s@#!/usr/bin/env sh@#!${bash}/bin/sh@g" \
-        -e "s@#!/usr/bin/env cat@#!${coreutils}/bin/cat@g"
-
-    sed -i tests/interrupts.rs \
-        -e "s@/bin/echo@${coreutils}/bin/echo@g" \
-        -e "s@#!/usr/bin/env sh@#!${bash}/bin/sh@g" \
-        -e "s@#!/usr/bin/env cat@#!${coreutils}/bin/cat@g"
+        -e "s@#!/usr/bin/env cat@#!${coreutils}/bin/cat@g" \
+        -e "s@#!/usr/bin/env bash@#!${bash}/bin/sh@g"
 
     sed -i src/justfile.rs \
         -e "s@/bin/echo@${coreutils}/bin/echo@g" \
         -e "s@#!/usr/bin/env sh@#!${bash}/bin/sh@g" \
         -e "s@#!/usr/bin/env cat@#!${coreutils}/bin/cat@g"
   '';
 
+  # Run "cargo test" without test case "edit",
+  # since test case "edit" needs "cat".
+  checkPhase = ''
+    runHook preCheck
+    echo "Running cargo cargo test
+        --lib
+        --test completions
+        --test init
+        --test integration
+        --test interrupts
+        --test invocation_directory
+        --test readme
+        --test search
+        --test shell
+        --test working_directory
+        -- ''${checkFlags} ''${checkFlagsArray+''${checkFlagsArray[@]}}"
+
+    cargo test \
+        --lib \
+        --test completions \
+        --test init \
+        --test integration \
+        --test interrupts \
+        --test invocation_directory \
+        --test readme \
+        --test search \
+        --test shell \
+        --test working_directory \
+        -- ''${checkFlags} ''${checkFlagsArray+''${checkFlagsArray[@]}}
+    runHook postCheck
+  '';
```

Can we just skip the failing case, instead of enumerating all the working ones?

```suggestion
  checkPhase = ''
    runHook preCheck
    echo "Running cargo test --
        --skip edit
        ''${checkFlags} ''${checkFlagsArray+''${checkFlagsArray[@]}}"
    cargo test -- \
        --skip edit \
        ''${checkFlags} ''${checkFlagsArray+"''${checkFlagsArray[@]}"}
    runHook postCheck
  '';
```